crates/webidl/src/idl_type.rs

```diff
@@ -88,6 +88,110 @@ pub(crate) enum IdlType<'a> {
     UnknownInterface(&'a str),
 }
 
+/// Indication that the [syn::Type] returned along with this does not tell the whole truth encoded
+/// in the IDL. All its properties are shaped by which information it needs in its `.tell()` method
+/// that converts the gathered information into something that can be put into the docs.
+#[derive(Debug)]
+pub struct Confession {
+    kind: ConfessionKind,
+    actual_type: Option<String>,
+    is_behind_result: bool,
+    has_more_data: bool,
+}
+
+#[derive(Debug)]
+enum ConfessionKind {
+    Promise,
+    Iterable,
+}
+
+impl Confession {
+    /// Create a textual description of what's wrong with the returned type.
+    ///
+    /// Any formatting in the return value is rustdoc-suitable Markdown.
+    pub fn tell(&self, in_return_position: bool) -> String {
+        (|| -> Result<_, std::fmt::Error> {
+            use std::fmt::Write;
+            let mut text = String::new();
+            let mut show_more = self.has_more_data;
+
+            match (&self.kind, &self.actual_type) {
+                (ConfessionKind::Promise, Some(s)) => {
+                    write!(text, "While the Promise can produce any JsValue as far as the type system is \
+                            concerned, practically it is expected to contain a `{}`.", s)?;
```

Comment:
Thing is, these can be more complex types as well (), and then this stops working; https://github.com/dtolnay/syn/issues/937 is trying to get the full solution. An easy 80% solution for 5% of the effort would be to make them links if they're alphanumeric only and not attempt anything fancy otherwise. 15% more for 5% more by stripping any whitespace for those composed of alphanumerics or `::` and making them links too. Unless https://github.com/dtolnay/syn/issues/937 comes around quickly, I'll pick one of those.

Reply:
Hm for the general case I could imagine that but for web-sys's use case we don't really have to deal with generics/lifetimes/etc, so I think the implementation should be pretty simple?

Reply:
The three simple cases are covered now: plain identifiers, Option<identifier> and &identifier.  My go-to pages to look at the visualization are Directory::get_files, CredentialsContainer::create and U2f::register to check them.

https://github.com/dtolnay/syn/issues/937 was closed as out-of-scope (which is OK, you might call the item a Zuständigkeitsfeststellungsfehlerbericht if you're into German composita); `doc_prettyprint` now goes as far as I consider it usable while documenting a bit of the general case this could be extended to if someone feels overly motivated.